### PR TITLE
Stop using unified memory by default

### PR DESCRIPTION
**Pull request by shinh**
Though we have not clarified the difference between unified
memory and normal GPU memory, a couple of performance
regressions were observed:

- A significant regression was observed when GPU memory usage
  is close to the limit. It looked like some GPU memory was
  migrated to the host even though GPU should have enough
  memory.
- A tiny regression was also observed for simple element-wise
  ops even when GPU memory usage is low.

The latter can be reproduced by

https://github.com/shinh/test/blob/master/unified_mem_perf.sh
https://github.com/shinh/test/blob/master/unified_mem_perf.py

on both p100 and v100.

We might eventually re-consider enabling unified memory, but
we are using no unified memory specific feature yet and I
guess this should be an opt-in feature like cupy's.

### CODE REVIEW

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
Please fix the logic in `CudaDeviceTest`.
Please also put a link to this PR there.

```
00:09:53.254 5: [ RUN      ] CudaDeviceTest.Allocate
00:09:53.255 5: /repo/chainerx_cc/chainerx/cuda/cuda_device_test.cc:54: Failure
00:09:53.255 5: Value of: IsPointerManagedMemory(ptr.get())
00:09:53.255 5:   Actual: false
00:09:53.255 5: Expected: true
```

**Comment by shinh** [comment]
I believe cuda tests were fixed, though I'm not 100% sure these tests are still appropriate. Suggestions are welcomed, thanks!

File: chainerx_cc/chainerx/cuda/cuda_backend_test.cc [hunk 1]
```diff
@@ -24,15 +24,26 @@ namespace chainerx {
 namespace cuda {
 namespace {
 
+template <typename T>
+std::shared_ptr<void> ToHost(const Device& from_device, const std::shared_ptr<void>& mem, size_t size) {
+    if (nullptr != dynamic_cast<native::NativeBackend*>(&from_device.backend())) return mem;
```
[the remark this answers is not in the document]
Comment (shinh) [reply chainerx_cc/chainerx/cuda/cuda_backend_test.cc]: Done

**Comment by shinh** [comment]
Thanks for the suggestions! I added a comment to DeviceMemoryAllocator::Alloc

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**